dcos/marathon.py

```diff
@@ -733,8 +722,29 @@ def get_leader(self):
         return response.json()['leader']
 
     def add_pod(self, pod_json):
+        """Add a new pod.
+
+        :param pod_json: JSON pod definition
+        :type pod_json: dict
+        :returns: description of created pod
+        :rtype: dict
+        """
         return self._rpc.http_req(http.post, 'v2/pods', json=pod_json)
 
+    def remove_pod(self, pod_id, force=False):
+        """Completely removes the requested pod.
+
+        :param pod_id: the ID of the pod to remove
+        :type pod_id: str
+        :param force: whether to override running deployments
+        :type force: bool
+        :rtype: None
+        """
+
+        path = 'v2/pods' + _normalize_marathon_id_path(pod_id)
```

Comment:
Argh, this is frustrating! The normalized path ID has a leading slash, which causes problems:

```python
os.path.join('v2/pods', '/foo/bar') == '/foo/bar'
os.path.join('v2/pods/', '/foo/bar') == '/foo/bar'
urllib.parse.urljoin('v2/pods', '/foo/bar') == '/foo/bar'
urllib.parse.urljoin('v2/pods/', '/foo/bar') == '/foo/bar'
```

I might just write a custom method for this.